armonik/locals.tf

```diff
@@ -55,182 +45,17 @@ locals {
   compute_plane_annotations = { for partition in local.partition_names : partition => try(var.compute_plane[partition].annotations, {}) }
   ingress_annotations       = try(var.ingress.annotations, {})
 
-  # Secrets
-  secrets = {
-    activemq = {
-      name        = "activemq"
-      ca_filename = "/amqp/chain.pem"
-    }
-    mongodb = {
-      name        = "mongodb"
-      ca_filename = "/mongodb/chain.pem"
-    }
-    redis = {
-      name        = "redis"
-      ca_filename = "/redis/chain.pem"
-    }
-    s3                             = var.s3_secret_name
-    shared_storage                 = var.shared_storage_secret_name
-    metrics_exporter               = var.metrics_exporter_secret_name
-    partition_metrics_exporter     = var.partition_metrics_exporter_secret_name
-    fluent_bit                     = var.fluent_bit_secret_name
-    seq                            = var.seq_secret_name
-    grafana                        = var.grafana_secret_name
-    prometheus                     = var.prometheus_secret_name
-    deployed_object_storage_secret = var.deployed_object_storage_secret_name
-    deployed_table_storage_secret  = var.deployed_table_storage_secret_name
-    deployed_queue_storage_secret  = var.deployed_queue_storage_secret_name
-  }
-
   # Shared storage
-  file_storage_type       = lower(data.kubernetes_secret.shared_storage.data.file_storage_type)
+  file_storage_type       = var.shared_storage_settings != null ? lower(var.shared_storage_settings.file_storage_type) : "FS"
   check_file_storage_type = local.file_storage_type == "s3" ? "S3" : "FS"
   file_storage_endpoints = local.check_file_storage_type == "S3" ? {
-    S3Storage__ServiceURL         = data.kubernetes_secret.shared_storage.data.service_url
-    S3Storage__AccessKeyId        = data.kubernetes_secret.shared_storage.data.access_key_id
-    S3Storage__SecretAccessKey    = data.kubernetes_secret.shared_storage.data.secret_access_key
-    S3Storage__BucketName         = data.kubernetes_secret.shared_storage.data.name
-    S3Storage__MustForcePathStyle = data.kubernetes_secret.shared_storage.data.must_force_path_style
-    S3Storage__UseChunkEncoding   = data.kubernetes_secret.shared_storage.data.use_chunk_encoding
-    S3Storage__UseChecksum        = data.kubernetes_secret.shared_storage.data.use_check_sum
+    S3Storage__ServiceURL         = var.shared_storage_settings.service_url
+    S3Storage__AccessKeyId        = var.shared_storage_settings.access_key_id
+    S3Storage__SecretAccessKey    = var.shared_storage_settings.secret_access_key
+    S3Storage__BucketName         = var.shared_storage_settings.name
+    S3Storage__MustForcePathStyle = var.shared_storage_settings.must_force_path_style
```

Comment:
This should most likely also go through the configuration mechanism.

Reply:
will be treated in a different PR.